dask/layers.py

```diff
@@ -423,6 +482,11 @@ def _dict(self):
             self._cached_dict = dsk
         return self._cached_dict
 
+    def __contains__(self, key):
```

Comment:
can you explain how this addition to `SimpleShuffleLayer` fits in with the rest of the PR?

Reply:
`SimpleShuffleLayer` has defined a `__getitem__` method which materializes the layer when only a single key is retrieved. So when something like `key in SimpleShuffleLayer` is tried, it will materialize the layer just to view membership - even though a check against the `get_output_keys()` value would suffice. 

So this modification allows `SimpleShuffleLayer` to remain unmaterialized when just checking if it contains a key. This was required for one of the tests, as it was looking at whether the layer was still unmaterialized after culling. 

Note that this issue might be present for other layers, if other layers don't have a similar test and become materialized from their `__getitem__` method.

Reply:
I think the way to fix that last point for all layer types would just be to define `__contains__` on the base `Layer` class, which checks against `get_output_keys`. As far as I understand, `get_output_keys` is the non-materialized equivalent to Layer.keys - so all inheriting layers should already be defining it in that way if possible.

Reply:
>This was required for one of the tests, as it was looking at whether the layer was still unmaterialized after culling.

Okay cool - That makes sense. Glad that we are catching this for at least one layer type.

>I think the way to fix that last point for all layer types would just be to define __contains__ on the base Layer class, which checks against get_output_keys.

Yeah, I think I like this idea. Let's try not to introduce any regressions like this that we are aware of.